### PR TITLE
Add 'HasNext', 'Next', and 'All' methods to PageIterator

### PR DESCRIPTION
**Pull request by eric-millin**
## Overview

Adds `HasNext`, `Next`, and `All` methods to `PageIterator`. These methods allow iteration without callbacks. 

## Testing

Unit tests included.

### CODE REVIEW

File: page_iterator.go [hunk 1]
```diff
@@ -125,20 +170,10 @@ func (pI *PageIterator[T]) GetOdataDeltaLink() *string {
...
-	page, err = convertToPage[T](resp)
-	if err != nil {
-		return page, err
-	}
-
-	return page, nil
+// HasNext returns true if there are additional items to iterate.
+func (pI *PageIterator[T]) HasNext() bool {
+	return pI.pauseIndex < len(pI.currentPage.getValue()) ||
+		pI.GetOdataNextLink() != nil && *pI.GetOdataNextLink() != ""
```
Comment (rkodev) [inline page_iterator.go:176]: defensive programing , we can check if the next link is a white space using `strings.TrimSpace(s)`

**Review by rkodev — CHANGES_REQUESTED** [review]
@eric-millin Thanks for the contribution.  Left a comment but overall this look good

**Review by rkodev — APPROVED** [review]

**Review by baywet — APPROVED** [review]
Thanks for the contribution!

**Comment by baywet** [comment]
@eric-millin would you mind also adding a entry to the changelog please? 
(bump the patch, use the date of tomorrow, and add your changes under an added section)
Thanks!

**Comment by baywet** [comment]
@eric-millin no, sorry I got mixed up with partial diffs in GitHub and forgot it was already there.
@rkodev for merge and publish

**Comment by eric-millin** [comment]
@baywet @rkodev There's an edge case bug during iteration. If page A has an `odata.next` field, but the following page has empty `values`, then `Next(ctx)` returns `nil`, which isn't what we want, I think.

I'm going to close this PR, fix it, and then can re-open or create a new PR.

**Pull request closed without merging**